src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -5,29 +5,35 @@ import { translationNotFoundMessage } from 'app/core/config/translation.config';
 import { EventManager, EventWithContent } from 'app/core/util/event-manager.service';
 import { AlertError } from 'app/shared/alert/alert-error.model';
 import { Subscription } from 'rxjs';
+import { captureException } from '@sentry/browser';
 
 export type AlertType = 'success' | 'danger' | 'warning' | 'info';
 
-export interface AlertBase {
+interface AlertBaseInternal {
     type: AlertType;
     message?: string;
-    translationKey?: string;
-    translationParams?: { [key: string]: unknown };
     timeout?: number;
-    action?: { label: string; callback: (alert: Alert) => void };
+    action?: { readonly label: string; readonly callback: (alert: Alert) => void };
     onClose?: (alert: Alert) => void;
     dismissible?: boolean;
 }
 
-export interface Alert extends AlertBase {
-    id: number;
-    close: () => void;
+export interface AlertCreationProperties extends Readonly<AlertBaseInternal> {
+    translationKey?: string;
+    translationParams?: { [key: string]: unknown };
+    disableTranslation?: boolean;
+}
+
+export interface Alert extends Readonly<AlertBaseInternal> {
+    readonly close: () => void;
     readonly isOpen: boolean;
 }
 
-export interface AlertInternal extends Alert {
+interface AlertInternal extends AlertBaseInternal {
```

Comment:
Shouldn't this rather extend `Alert`?

Reply:
No, ``Alert`` is the public readonly interface. Internally, we want to be able to overwrite fields (which we do a lot in the lines below). That's why ``close`` and ``isOpen`` are defined in both interfaces (one time as readonly, one time as regular field)